src/runtime/instruction/areturn.rs

```diff
@@ -11,42 +11,30 @@ pub fn areturn(_vm: &Vm, ctx: &mut CallSite, _bytes: &mut Bytes) -> Result<()> {
     let mut sf = lock.pop().expect("call stack was empty?");
 
     let ref_value = sf.operand_stack.pop();
+    let ref_value = ref_value.ok_or_else(|| anyhow!("operand stack was empty"))?;
 
-    if ref_value.is_none() {
-        return Err(anyhow!("operand stack was empty"));
-    }
-
+    // TODO: I don't quite understand this, "operand stack was empty" is an error, and operand stack not empty is a warn?
```

Comment:
Reporting runtime errors is still to be designed, though I an clear up confusion. `areturn` clears all of the values on the stack (as it returns) but if there's values on the stack already that would indicate invalid code. 

This warns as its recoverable, where returning an error from an instruction indicates that the VM has to halt. Ideally we should have error types where the caller can decide how to proceed based on the type.